Build failed on macOS Sieera

When I tried to compile by just following the instruction in README on my mac,
I encountered the following error and I'm sorry I'm not quite sure about this message.
Is there anyone having seen this??
I had also tried it with Ninja, but it failed with kind of different error neither.
I past both cases below.

```
# make
>> cmake .. -DCMAKE_BUILD_TYPE=Debug -DGLOW_WITH_CPU=1 -DGLOW_WITH_OPENCL=0 -DCMAKE_PREFIX_PATH=/usr/local/opt/llvm@6/
-- The C compiler identification is AppleClang 9.1.0.9020039
-- The CXX compiler identification is AppleClang 9.1.0.9020039
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Tests not explicitly enabled/disabled, default to enable
-- Found ZLIB: /usr/lib/libz.dylib (found version "1.2.11")
-- Found PNG: /usr/local/lib/libpng.dylib (found version "1.4.12")
CMake Warning (dev) at CMakeLists.txt:61 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Environment variable LLVM_ROOT is set to:

    /usr/local/opt/llvm@6

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found LLVM 6.0.1
-- Using LLVMConfig.cmake in: /usr/local/opt/llvm@6/lib/cmake/llvm
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Found Protobuf: /usr/local/lib/libprotobuf.dylib (found version "3.6.0")
-- Found PythonInterp: /Users/yamada/.pyenv/shims/python (found version "3.6.3")
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/yamada/tk-home/ghq/github.com/pytorch/glow/build_Debug
>> make -j4                                                             
Scanning dependencies of target QuantizationBase
Scanning dependencies of target Support
Scanning dependencies of target CPURuntime
Scanning dependencies of target CodeGen
[  1%] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntime.dir/libjit/libjit.cpp.o
[  2%] Building CXX object lib/CodeGen/CMakeFiles/CodeGen.dir/MemoryAllocator.cpp.o
[  2%] Building CXX object lib/Quantization/Base/CMakeFiles/QuantizationBase.dir/Profile.cpp.o
[  3%] Building CXX object lib/Support/CMakeFiles/Support.dir/Debug.cpp.o
[  3%] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntime.dir/libjit/libjit_conv.cpp.o
[  4%] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntime.dir/libjit/libjit_matmul.cpp.o
[  4%] Linking CXX shared library liblibjit.bc.dylib
llvm-link: Unknown command line argument '-Wall'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-help'?
llvm-link: Unknown command line argument '-Wnon-virtual-dtor'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-print-after'?
llvm-link: Unknown command line argument '-fno-exceptions'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-print-options'?
llvm-link: Unknown command line argument '-fno-rtti'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-rng-seed'?
llvm-link: Unknown command line argument '-g'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-S'?
llvm-link: Unknown command line argument '-fno-omit-frame-pointer'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-import-all-index'?
llvm-link: Unknown command line argument '-O0'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-S'?
llvm-link: Unknown command line argument '-dynamiclib'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-rng-seed'?
llvm-link: Unknown command line argument '-Wl,-headerpad_max_install_names'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-as-secure-log-file-name'?
llvm-link: for the -o option: may only occur zero or one times!
llvm-link: Unknown command line argument '-install_name'.  Try: '/usr/local/opt/llvm@6/bin/llvm-link -help'
llvm-link: Did you mean '-internalize'?
make[2]: *** [lib/Backends/CPU/liblibjit.bc.dylib] Error 1
make[1]: *** [lib/Backends/CPU/CMakeFiles/CPURuntime.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  6%] Building CXX object lib/Quantization/Base/CMakeFiles/QuantizationBase.dir/Base.cpp.o
[  6%] Building CXX object lib/Support/CMakeFiles/Support.dir/Random.cpp.o
[  7%] Building CXX object lib/Support/CMakeFiles/Support.dir/Support.cpp.o
[  7%] Linking CXX static library libQuantizationBase.a
[  7%] Built target QuantizationBase
[  7%] Linking CXX static library libCodeGen.a
[  7%] Built target CodeGen
[  7%] Linking CXX static library libSupport.a
[  7%] Built target Support
make: *** [all] Error 2
```

```
# Ninja
>> ninja all   
[9/159] Running gen_proto.py on onnx/onnx.in.proto
Processing /Users/yamada/tk-home/ghq/github.com/pytorch/glow/thirdparty/onnx/onnx/onnx.in.proto
Writing /Users/yamada/tk-home/ghq/github.com/pytorch/glow/build_Debug/lib/Importer/build/onnx/onnx_glow_onnx.proto
Writing /Users/yamada/tk-home/ghq/github.com/pytorch/glow/build_Debug/lib/Importer/build/onnx/onnx_glow_onnx.proto3
Writing /Users/yamada/tk-home/ghq/github.com/pytorch/glow/build_Debug/lib/Importer/build/onnx/onnx.pb.h
generating /Users/yamada/tk-home/ghq/github.com/pytorch/glow/build_Debug/lib/Importer/build/onnx/onnx_pb.py
[15/159] Building CXX object lib/Base/CMakeFiles/Base.dir/Tensor.cpp.o
FAILED: lib/Base/CMakeFiles/Base.dir/Tensor.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DGLOW_WITH_CPU=1 -DWITH_PNG=1 -I../include -Iinclude -I. -I/Library/Frameworks/Mono.framework/Headers -isystem /usr/local/opt/llvm@6/include -Wall -Wnon-virtual-dtor -fno-exceptions -fno-rtti -g -fno-omit-frame-pointer -O0 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=c++14 -MD -MT lib/Base/CMakeFiles/Base.dir/Tensor.cpp.o -MF lib/Base/CMakeFiles/Base.dir/Tensor.cpp.o.d -o lib/Base/CMakeFiles/Base.dir/Tensor.cpp.o -c ../lib/Base/Tensor.cpp
In file included from ../lib/Base/Tensor.cpp:17:
../include/glow/Base/Tensor.h:299:5: error: use of undeclared identifier 'llvm_unreachable'
    llvm_unreachable("unreachable");
    ^
../lib/Base/Tensor.cpp:188:5: error: use of undeclared identifier 'llvm_unreachable'
    llvm_unreachable("Invalid tensor size");
    ^
2 errors generated.
[16/159] Building CXX object lib/Base/CMakeFiles/Base.dir/Image.cpp.o
FAILED: lib/Base/CMakeFiles/Base.dir/Image.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DGLOW_WITH_CPU=1 -DWITH_PNG=1 -I../include -Iinclude -I. -I/Library/Frameworks/Mono.framework/Headers -isystem /usr/local/opt/llvm@6/include -Wall -Wnon-virtual-dtor -fno-exceptions -fno-rtti -g -fno-omit-frame-pointer -O0 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=c++14 -MD -MT lib/Base/CMakeFiles/Base.dir/Image.cpp.o -MF lib/Base/CMakeFiles/Base.dir/Image.cpp.o.d -o lib/Base/CMakeFiles/Base.dir/Image.cpp.o -c ../lib/Base/Image.cpp
In file included from ../lib/Base/Image.cpp:17:
In file included from ../include/glow/Base/Image.h:19:
../include/glow/Base/Tensor.h:299:5: error: use of undeclared identifier 'llvm_unreachable'
    llvm_unreachable("unreachable");
    ^
1 error generated.
[17/159] Running gen_proto.py on onnx/onnx-operators.in.proto
Processing /Users/yamada/tk-home/ghq/github.com/pytorch/glow/thirdparty/onnx/onnx/onnx-operators.in.proto
Writing /Users/yamada/tk-home/ghq/github.com/pytorch/glow/build_Debug/lib/Importer/build/onnx/onnx-operators_glow_onnx.proto
Writing /Users/yamada/tk-home/ghq/github.com/pytorch/glow/build_Debug/lib/Importer/build/onnx/onnx-operators_glow_onnx.proto3
Writing /Users/yamada/tk-home/ghq/github.com/pytorch/glow/build_Debug/lib/Importer/build/onnx/onnx-operators.pb.h
generating /Users/yamada/tk-home/ghq/github.com/pytorch/glow/build_Debug/lib/Importer/build/onnx/onnx_operators_pb.py
[24/159] Building CXX object tests/unittests/CMakeFiles/tensorsTest.dir/tensorsTest.cpp.o
ninja: build stopped: subcommand failed.
```

```
 grokmachine@spock   ~/dev/glow/build     master  cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH=/usr/local/opt/llvm ..                                                                                                                ✔  10290  13:40:53
-- The C compiler identification is Clang 6.0.1
-- The CXX compiler identification is Clang 6.0.1
-- Check for working C compiler: /usr/local/opt/llvm/bin/clang
-- Check for working C compiler: /usr/local/opt/llvm/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++
-- Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Tests not explicitly enabled/disabled, default to enable
-- Found ZLIB: /usr/lib/libz.dylib (found version "1.2.11")
-- Found PNG: /usr/local/lib/libpng.dylib (found version "1.6.35")
-- Looking for CL_VERSION_2_2
-- Looking for CL_VERSION_2_2 - not found
-- Looking for CL_VERSION_2_1
-- Looking for CL_VERSION_2_1 - not found
-- Looking for CL_VERSION_2_0
-- Looking for CL_VERSION_2_0 - not found
-- Looking for CL_VERSION_1_2
-- Looking for CL_VERSION_1_2 - found
-- Found OpenCL: /System/Library/Frameworks/OpenCL.framework (found version "1.2")
-- Found LLVM 6.0.1
-- Using LLVMConfig.cmake in: /usr/local/opt/llvm/lib/cmake/llvm
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Found Protobuf: /usr/local/lib/libprotobuf.dylib (found version "3.4.0")
-- Found PythonInterp: /Users/grokmachine/.pyenv/shims/python (found version "3.6.2")
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/grokmachine/dev/glow/build
 grokmachine@spock   ~/dev/glow/build     master  make -j20                                                                                                                                                                                                                                      ✔  10291  13:41:03
Scanning dependencies of target CPURuntimeNative
Scanning dependencies of target gen_onnx_proto
Scanning dependencies of target Support
Scanning dependencies of target QuantizationBase
Scanning dependencies of target CPURuntime
Scanning dependencies of target CodeGen
Scanning dependencies of target gmock
Scanning dependencies of target gtest
Scanning dependencies of target gmock_main
[  0%] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntime.dir/libjit/libjit_matmul.cpp.o
[  0%] Building CXX object lib/Quantization/Base/CMakeFiles/QuantizationBase.dir/Profile.cpp.o
[  1%] Building CXX object lib/Quantization/Base/CMakeFiles/QuantizationBase.dir/Base.cpp.o
[  2%] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntime.dir/libjit/libjit_conv.cpp.o
[  2%] Building CXX object lib/CodeGen/CMakeFiles/CodeGen.dir/MemoryAllocator.cpp.o
[  3%] Building CXX object lib/Support/CMakeFiles/Support.dir/Debug.cpp.o
[  3%] Building CXX object lib/Support/CMakeFiles/Support.dir/Random.cpp.o
[  4%] Building CXX object lib/Support/CMakeFiles/Support.dir/Support.cpp.o
[  6%] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntimeNative.dir/libjit/libjit_conv.cpp.o
[  6%] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntimeNative.dir/libjit/libjit.cpp.o
[  7%] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntimeNative.dir/libjit/libjit_matmul.cpp.o
[  7%] Running gen_proto.py on onnx/onnx.in.proto
[  7%] Building CXX object lib/Backends/CPU/CMakeFiles/CPURuntime.dir/libjit/libjit.cpp.o
[  7%] Building CXX object tests/googletest/googlemock/CMakeFiles/gmock.dir/__/googletest/src/gtest-all.cc.o
[  8%] Building CXX object tests/googletest/googlemock/CMakeFiles/gmock.dir/src/gmock-all.cc.o
[  9%] Building CXX object tests/googletest/googlemock/CMakeFiles/gmock_main.dir/__/googletest/src/gtest-all.cc.o
[  9%] Building CXX object tests/googletest/googlemock/CMakeFiles/gmock_main.dir/src/gmock-all.cc.o
[ 10%] Building CXX object tests/googletest/googlemock/CMakeFiles/gmock_main.dir/src/gmock_main.cc.o
[ 12%] Building CXX object tests/googletest/googlemock/gtest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
Processing /Users/grokmachine/dev/glow/thirdparty/onnx/onnx/onnx.in.proto
Writing /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx_glow_onnx.proto
Writing /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx_glow_onnx.proto3
Writing /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx.pb.h
generating /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx_pb.py
[ 12%] Running C++ protocol buffer compiler on /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx_glow_onnx.proto
[ 12%] Built target gen_onnx_proto
[ 13%] Running gen_proto.py on onnx/onnx-operators.in.proto
Processing /Users/grokmachine/dev/glow/thirdparty/onnx/onnx/onnx-operators.in.proto
Writing /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx-operators_glow_onnx.proto
Writing /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx-operators_glow_onnx.proto3
Writing /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx-operators.pb.h
generating /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx_operators_pb.py
[ 14%] Running C++ protocol buffer compiler on /Users/grokmachine/dev/glow/build/lib/Importer/build/onnx/onnx-operators_glow_onnx.proto
Scanning dependencies of target onnx_proto
[ 15%] Linking CXX shared library liblibjit.bc.dylib
llvm-link: Unknown command line argument '-Wall'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-help'?
llvm-link: Unknown command line argument '-Wnon-virtual-dtor'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-unroll-remainder'?
llvm-link: Unknown command line argument '-fno-exceptions'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-print-options'?
llvm-link: Unknown command line argument '-fno-rtti'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-rng-seed'?
llvm-link: Unknown command line argument '-g'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-S'?
llvm-link: Unknown command line argument '-fno-omit-frame-pointer'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-spp-print-base-pointers'?
llvm-link: Unknown command line argument '-O0'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-S'?
llvm-link: Unknown command line argument '-dynamiclib'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-dfsan-abilist'?
llvm-link: Unknown command line argument '-Wl,-headerpad_max_install_names'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-memop-max-annotations'?
llvm-link: for the -o option: may only occur zero or one times!
llvm-link: Unknown command line argument '-install_name'.  Try: '/usr/local/opt/llvm/bin/llvm-link -help'
llvm-link: Did you mean '-internalize'?
make[2]: *** [lib/Backends/CPU/liblibjit.bc.dylib] Error 1
make[1]: *** [lib/Backends/CPU/CMakeFiles/CPURuntime.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 15%] Building CXX object lib/Importer/build/CMakeFiles/onnx_proto.dir/onnx/onnx_glow_onnx.pb.cc.o
[ 17%] Building CXX object lib/Importer/build/CMakeFiles/onnx_proto.dir/onnx/onnx-operators_glow_onnx.pb.cc.o
[ 18%] Linking CXX static library libCPURuntimeNative.a
[ 18%] Built target CPURuntimeNative
[ 18%] Linking CXX static library libQuantizationBase.a
[ 18%] Linking CXX static library libSupport.a
[ 18%] Built target QuantizationBase
[ 18%] Built target Support
[ 19%] Linking CXX static library libCodeGen.a
[ 19%] Built target CodeGen
[ 19%] Linking CXX static library libonnx_proto.a
[ 19%] Built target onnx_proto
[ 19%] Linking CXX static library libgmock.a
[ 19%] Linking CXX static library libgmock_main.a
[ 19%] Linking CXX static library libgtest.a
[ 19%] Built target gtest
[ 19%] Built target gmock
[ 19%] Built target gmock_main
make: *** [all] Error 2
```

--- a/lib/Backends/CPU/CMakeLists.txt
+++ b/lib/Backends/CPU/CMakeLists.txt
@@ -76,24 +76,24 @@ target_link_libraries(CPUBackend
                         IR
                         Optimizer
                         QuantizationBase
+                        ${LLVM_TARGET_LIBRARIES}
                         LLVMAnalysis
+                        LLVMBitWriter
                         LLVMCodeGen
                         LLVMCore
-                        LLVMipo
+                        LLVMExecutionEngine
                         LLVMIRReader
                         LLVMInstCombine
+                        LLVMInterpreter
                         LLVMMC
+                        LLVMObject
+                        LLVMPasses
                         LLVMScalarOpts
                         LLVMSupport
                         LLVMTarget
                         LLVMTransformUtils
                         LLVMVectorize
-                        ${LLVM_TARGET_LIBRARIES}
-                        LLVMCore
-                        LLVMExecutionEngine
-                        LLVMInterpreter
-                        LLVMSupport
-                        LLVMPasses)
+                        LLVMipo)
 if(LLVM_VERSION_MAJOR VERSION_GREATER 6)
   target_link_libraries(CPUBackend
                         PRIVATE

